_data/engine-cli/docker_checkpoint.yaml

```diff
@@ -1,6 +1,98 @@
 command: docker checkpoint
 short: Manage checkpoints
-long: Manage checkpoints
+long: |-
+  Checkpoint and Restore is an experimental feature that allows you to freeze a running
+  container by checkpointing it, which turns its state into a collection of files
+  on disk. Later, the container can be restored from the point it was frozen.
+
+  This is accomplished using a tool called [CRIU](http://criu.org), which is an
+  external dependency of this feature. A good overview of the history of
+  checkpoint and restore in Docker is available in this
+  [Kubernetes blog post](https://kubernetes.io/blog/2015/07/how-did-quake-demo-from-dockercon-work/).
+
+  ### Installing CRIU
+
+  If you use a Debian system, you can add the CRIU PPA and install with apt-get
+  [from the criu launchpad](https://launchpad.net/~criu/+archive/ubuntu/ppa).
+
+  Alternatively, you can [build CRIU from source](https://criu.org/Installation).
+
+  You need at least version 2.0 of CRIU to run checkpoint/restore in Docker.
+
+  ### Use cases for checkpoint & restore
+
+  This feature is currently focused on single-host use cases for checkpoint and
+  restore. Here are a few:
+
+  - Restarting the host machine without stopping/starting containers
+  - Speeding up the start time of slow start applications
+  - "Rewinding" processes to an earlier point in time
+  - "Forensic debugging" of running processes
+
+  Another primary use case of checkpoint & restore outside of Docker is the live
+  migration of a server from one machine to another. This is possible with the
+  current implementation, but not currently a priority (and so the workflow is
+  not optimized for the task).
+
+  ### Using checkpoint & restore
+
+  A new top level command `docker checkpoint` is introduced, with three subcommands:
+  - `create` (creates a new checkpoint)
+  - `ls` (lists existing checkpoints)
+  - `rm` (deletes an existing checkpoint)
+
+  Additionally, a `--checkpoint` flag is added to the container start command.
+
+  The options for checkpoint create:
+
+  ```console
+  Usage:  docker checkpoint create [OPTIONS] CONTAINER CHECKPOINT
+
+  Create a checkpoint from a running container
+
+    --leave-running=false    Leave the container running after checkpoint
+    --checkpoint-dir         Use a custom checkpoint storage directory
+  ```
+
+  And to restore a container:
```

Comment:
```suggestion
  To restore a container:
```

Reply:
Ah, looks like I missed this one; will keep that for a follow-up